Find a functionally equivalent code:android.print.PrintManager.addPrintJobStateChangeListener:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * Adds a listener for observing the state of print jobs.
 *
 * @param listener The listener to add.
 * @hide
 */

@@ -1,4 +1,8 @@
 {
+    if (mService == null) {
+        Log.w(LOG_TAG, "Feature android.software.print not available");
+        return;
+    }
     if (mPrintJobStateChangeListeners == null) {
         mPrintJobStateChangeListeners = new ArrayMap<PrintJobStateChangeListener, PrintJobStateChangeListenerWrapper>();
     }

